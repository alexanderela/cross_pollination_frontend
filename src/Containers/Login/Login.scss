$blue-accent: rgba(89, 219, 230, 1);
// blue-accent HEX: #59DBE6
$btn-white: #e2e2e2;
$fb-blue: #3b5998;
$tw-blue: #1dcaff;

* {
  transition-duration: .2s;
}

.Login {
  background: left / cover url(../../images/intros/flagGrid.png);
  background-repeat: repeat;
  background-position: center;
  height: 100vh;
  animation: flag-grid-slider 420s linear infinite;
}

.login-background-color {
  background: linear-gradient(rgba(12, 0, 254, .90) 3%, rgba(0, 0, 0, .90));
  height: 100%;
  width: 100%;
}

.login-title {
  color: #e2e2e2;
  font-family: 'Lobster';
  font-size: 4rem;
  letter-spacing: 2px;
  margin: 0;
  padding: 40% 0 20%;
  text-align: center;
  text-decoration: underline;
  width: 100%;
}

.login-title-small {
  color: #e2e2e2;
  font-family: 'Lobster';
  font-size: 2rem;
  letter-spacing: 2px;
  margin: 0;
  padding: 1.5rem 0;
  text-align: center;
  text-decoration: underline;
  width: 100%;
}

.login-divider {
  display: inline-flex;
  justify-content: space-between;
  height: 5rem;
  margin: 2rem 20%;
  text-align: center;
  width: 60%;
}

.divider {
  background-color: #59DBE6;
  height: 1px;
  margin: 2.5rem 0;
  width: 3rem;
}

.login-divider-text {
  color: #59DBE6;
  font-size: 1.5rem;
  letter-spacing: -2px;
  margin: 1.5rem 0;
}

.login-button-facebook {
  background-color: $fb-blue;
  color: $btn-white;
}

.login-button-twitter {
  background-color: $tw-blue;
}

.login-button-email {
  background-color: $btn-white;
}

.login-button-email,
.login-button-facebook,
.login-button-twitter,
.login-input {
  border-radius:3rem;
  box-shadow: 0 2px 8px rgba(0, 0, 0, .3);
  font-weight: 700;
  margin: 1.25rem auto;
  padding: 12px 0;
  text-align: center;
  transition-duration: .05s;
  width: 80%;
}

.login-back {
  border: none;
  border-radius:3rem;
  box-shadow: 0 2px 8px rgba(0, 0, 0, .3);
  font-size: 1rem;
  font-weight: 700;
  margin: 1.25rem auto;
  outline: none;
  padding: 12px 0;
  text-align: center;
  transition-duration: .05s;
  width: 100%;
}

.login-form {
  padding: 0 10%;
  width: 80%;
}

.form-instructions {
  color: $blue-accent;
  font-weight: 700;
  margin: 1rem 0;
  text-align: center;
  width: 100%;
}

.login-signup-slider {
  color: $blue-accent;
  display: inline-flex;
  font-size: 1.5rem;
  justify-content: space-around;
  margin: 1rem auto;
  width: 100%;
}

.form-slider {
  border: 2px solid $blue-accent;
  border-radius: 1rem;
  bottom: 2px;
  height: 2rem;
  position: relative;
  transition-duration: .2s;
  width: 35%;
}

.form-slider-login {
  right: 27%;
}

.form-slider-signup {
  left: 31.5%;
}

.form-slider-login,
.form-slider-signup {
  width: 30%;
}

.form-slider-login {
  text-align: right;
}

.form-slider-signup {
  text-align: left;
}

.login-input {
  border: none;
  font-size: 1rem;
  margin: 0 0 1.5rem 0;
  outline: none;
  text-align: left;
  text-indent: 1.5rem;
  width: 100%;
}

.login-input-placeholder {
  bottom: 3.5rem;
  height: 0;
  left: 1.5rem;
  color: black;
  position: relative;
  transition-duration: .2s;
}

.login-name:focus + .name-placeholder,
.login-email:focus + .email-placeholder,
.login-password:focus + .password-placeholder {
  bottom: 4.25rem;
  color: $fb-blue;
  font-weight: 700;
  font-size: 10px;
}


@keyframes flag-grid-slider {
  0% {
    background-position: 0 0;
  }

  100% {
    background-position: 1000% 0;
  }
}