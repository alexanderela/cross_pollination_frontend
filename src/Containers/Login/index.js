import React, { Component } from 'react';
import { Route, NavLink } from 'react-router-dom';
import * as API from '../../utilities/API';
import { connect } from 'react-redux';
import { fetchUser } from '../../Thunks/user';
import './Login.scss';
// import city1 from '../../images/intros/b.png'

class Login extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: '',
      createUser: false,
      error: '',
      emailCredentials: false,
      formLogin: true,
    }
  }

  createUser = () => {
    this.setState({createUser: !this.state.createUser});
  }

  loginUser = async (event) => {
    const { createUser, name, email, password } = this.state;
    event.preventDefault();
    return createUser
      ? this.props.fetchUser(name, email, password)
      : this.props.fetchUser(null, email, password) 
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  }

  expandCredentials = () => {
    this.setState({
      emailCredentials: true
    });
  }

  closeCredentials = () => {
    this.setState({
      emailCredentials: false
    });
  }

  changeFormPurpose = () => {
    const { formLogin } = this.state;
    this.setState({
      formLogin: !formLogin
    })
  }

  // submitLogin = async (e) => {
  //   e.preventDefault();
  //   const { email, password } = this.state;
  // };

  handleSubmit = async (event) => {
    await this.loginUser(event)
    if (this.props.loading !== `Email & password don't match`) {
      this.props.activateLogin();
    }
  }

  clearInputs = () => {
    this.setState({
      name: '',
      email: '',
      password: '',
    });
  }
;
  render() {
    const { emailCredentials, formLogin, name, email, password } = this.state;
    const { loading } = this.props;
    const showError = loading === `Email & password don't match` || loading === `Login to save score`
      ? loading
      : ''

    return (
      <div className='Login'>
        <div className='login-background-color'>
        <h1 className={!emailCredentials ? 'login-title' : 'login-title-small'}>Flagship</h1>
          {
            !emailCredentials &&
            <div className='login-divider'>
              <div className='divider'></div>
              <div className='login-divider-text'>login / sign up</div>
              <div className='divider'></div>
            </div>
          }
          <div className='login-button-facebook'>Facebook</div>
          <div className='login-button-twitter'>Twitter</div>
          {
            !emailCredentials &&
            <div className='login-button-email' onClick={this.expandCredentials}>Email</div>
          }
          {
            emailCredentials &&
            <form className='login-form'>
              <div className='form-instructions'>press return to submit</div>
              <div className='login-signup-slider' onClick={this.changeFormPurpose}>
                <div className='form-slider-login'>login</div>
                <div className={formLogin ? 'form-slider form-slider-login' : 'form-slider form-slider-signup'}></div>
                <div className='form-slider-signup'>sign up</div>
              </div>
              {
                !formLogin &&
                <div>
                  <input className='login-input login-name' name='name' />
                  <div className='login-input-placeholder name-placeholder'>name</div>
                </div>
              }
              <input className='login-input login-email' name='email' />
              <div className='login-input-placeholder email-placeholder'>email</div>
              <input className='login-input login-password' name='password' />
              <div className='login-input-placeholder password-placeholder'>password</div>
              <button className='login-back' onClick={this.closeCredentials}>go back</button>
            </form>
          }
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  loading: state.loading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchUser:(name, email, password) => dispatch(fetchUser(name, email, password))
})

export default connect (mapStateToProps, mapDispatchToProps)(Login);
