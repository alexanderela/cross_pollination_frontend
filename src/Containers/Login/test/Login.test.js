import React from 'react';
import { shallow, mount } from 'enzyme';
import Login from '../index';

describe('Login', () => {
  let wrapper;
  let mockEvent;
  let mockFunc;
  
  beforeEach(() => {
    mockFunc = jest.fn();
    mockEvent = {
      preventDefault: mockFunc,
      target: {
        name: 'name',
        value: 'Alex'
      }
    }
    wrapper = shallow(<Login loginUser={jest.fn()} />);
  });
  
    it('should render like the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
  describe('handleChange', () => {
    it.skip('should set state upon invocaton of handleChange', () => {
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state().name).toEqual('Alex')
    });
  });

  describe('submitLogin', () => {
    it.skip('should invoke loginUser if loginAttempt is successful', async () => {
      await wrapper.instance().submitLogin(mockEvent);
      expect(wrapper.instance().props.loginUser).toHaveBeenCalled();
    });
  });

  describe('createNewUser', () => {
    it('should alert the user if the account already exists', () => {
    });

    it('should set state for "create" or "error" if create state is false and fetchResponse is successful', () => {
    });

    it('should should set "error" state if "create" state is true and fetch response fails', () => {
    });

    it('should clear inputs if "create" state is true and fetch response fails', () => {
    });
  });
  
  describe('newUserResponse', () => {
    it('should invoke API.createUser upon invocation', () => {
    });
  });
  
  describe('clearInputs', () => {
    it.skip('should clear inputs on submit', () => {
      wrapper.setState({
        name: 'Bruce',
        email: 'Ela',
        password: 'Rau',
      });
      wrapper.instance().clearInputs();
      expect(wrapper.state().name).toEqual('');
      expect(wrapper.state().email).toEqual('');
      expect(wrapper.state().password).toEqual('');
    });
  });

  describe('mapStateToProps', () => {
    it('should create the correct props object', () => {
    });

  });
  
  describe('mapDispatchToProps', () => {
    it('should map a key of setCountries', () => {
    });
    
    it('setCountries should call dispatch', () => {
    });
  });
});

it('should', () => {});