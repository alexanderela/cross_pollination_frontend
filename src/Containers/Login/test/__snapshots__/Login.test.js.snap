// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render like the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    fetchUser={[MockFunction]}
    loading="true"
    user={
      Object {
        "email": "alex@turing.com",
        "id": 2,
        "loggedIn": true,
        "name": "rtg",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="login-background-color"
      >
        <h1
          className="login-title"
        >
          World
          <br />
          Of Flags
        </h1>
        <div
          className="login-divider"
        >
          <div
            className="divider"
          />
          <div
            className="login-divider-text"
          >
            login / sign up
          </div>
          <div
            className="divider"
          />
        </div>
        <div
          className="login-button-facebook"
        >
          Facebook
        </div>
        <div
          className="login-button-twitter"
        >
          Twitter
        </div>
        <div
          className="login-button-email"
          onClick={[Function]}
        >
          Email
        </div>
      </div>,
      "className": "Login",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="login-title"
          >
            World
            <br />
            Of Flags
          </h1>,
          <div
            className="login-divider"
          >
            <div
              className="divider"
            />
            <div
              className="login-divider-text"
            >
              login / sign up
            </div>
            <div
              className="divider"
            />
          </div>,
          <div
            className="login-button-facebook"
          >
            Facebook
          </div>,
          <div
            className="login-button-twitter"
          >
            Twitter
          </div>,
          <div
            className="login-button-email"
            onClick={[Function]}
          >
            Email
          </div>,
          false,
        ],
        "className": "login-background-color",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "World",
              <br />,
              "Of Flags",
            ],
            "className": "login-title",
          },
          "ref": null,
          "rendered": Array [
            "World",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Of Flags",
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="divider"
              />,
              <div
                className="login-divider-text"
              >
                login / sign up
              </div>,
              <div
                className="divider"
              />,
            ],
            "className": "login-divider",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "login / sign up",
                "className": "login-divider-text",
              },
              "ref": null,
              "rendered": "login / sign up",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "divider",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Facebook",
            "className": "login-button-facebook",
          },
          "ref": null,
          "rendered": "Facebook",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Twitter",
            "className": "login-button-twitter",
          },
          "ref": null,
          "rendered": "Twitter",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Email",
            "className": "login-button-email",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Email",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="login-background-color"
        >
          <h1
            className="login-title"
          >
            World
            <br />
            Of Flags
          </h1>
          <div
            className="login-divider"
          >
            <div
              className="divider"
            />
            <div
              className="login-divider-text"
            >
              login / sign up
            </div>
            <div
              className="divider"
            />
          </div>
          <div
            className="login-button-facebook"
          >
            Facebook
          </div>
          <div
            className="login-button-twitter"
          >
            Twitter
          </div>
          <div
            className="login-button-email"
            onClick={[Function]}
          >
            Email
          </div>
        </div>,
        "className": "Login",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="login-title"
            >
              World
              <br />
              Of Flags
            </h1>,
            <div
              className="login-divider"
            >
              <div
                className="divider"
              />
              <div
                className="login-divider-text"
              >
                login / sign up
              </div>
              <div
                className="divider"
              />
            </div>,
            <div
              className="login-button-facebook"
            >
              Facebook
            </div>,
            <div
              className="login-button-twitter"
            >
              Twitter
            </div>,
            <div
              className="login-button-email"
              onClick={[Function]}
            >
              Email
            </div>,
            false,
          ],
          "className": "login-background-color",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "World",
                <br />,
                "Of Flags",
              ],
              "className": "login-title",
            },
            "ref": null,
            "rendered": Array [
              "World",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Of Flags",
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="divider"
                />,
                <div
                  className="login-divider-text"
                >
                  login / sign up
                </div>,
                <div
                  className="divider"
                />,
              ],
              "className": "login-divider",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "divider",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "login / sign up",
                  "className": "login-divider-text",
                },
                "ref": null,
                "rendered": "login / sign up",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "divider",
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Facebook",
              "className": "login-button-facebook",
            },
            "ref": null,
            "rendered": "Facebook",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Twitter",
              "className": "login-button-twitter",
            },
            "ref": null,
            "rendered": "Twitter",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Email",
              "className": "login-button-email",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Email",
            "type": "div",
          },
          false,
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
